### ISSUE TITLE
TIFF files with “packbits compression” cannot be converted to PDF files

### ISSUE THREAD

**Issue by leo20160311** [pymupdf/PyMuPDF#2320]
mydoc = fitz.open(stream=tiff_bytes, filetype=‘tif’)

    mydoc = mydoc.convert_to_pdf()
    mydoc = fitz.open('pdf', mydoc)

    outdoc = fitz.open()
    outdoc.insert_pdf(mydoc)
    outdoc.save(outpdfpath)

Can someone help ？   generated file content is incorrect

**Comment by JorjMcKie** [comment]
Please provide a file to reproduce.

**Comment by JorjMcKie** [comment]
If you can open the TIFF as a `Pixmap`, you can still insert it as an image in a PDF page of suitable size:
```python
tif = "file.tif"
pix = fitz.Pixmap(tif)
doc=fitz.open()
page=doc.new_page(width=pix.width, height=pix.height)
page.insert_image(page.rect, filename=tif)
doc.save("file.pdf",deflate=True)
```

**Comment by leo20160311** [comment]
@JorjMcKie Thank you！

I tested more than ten TIFF files, but none of them could be converted to the correct PDF. The content of the converted PDF file is black.

Additionally, I am unable to upload files on Github and have already uploaded them to Gitee:
https://gitee.com/liudg2012/tiff.git

**Comment by leo20160311** [comment]
@JorjMcKie Thank You！
Additionally, in practical use, my TIFF file is multi page and using pixmap is not very convenient

**Comment by JorjMcKie** [comment]
> Additionally, I am unable to upload files on Github and have already uploaded them to Gitee

Sorry cannot find any graphic file there. Please use a conventional ZIP of at least one TIFF.
BTW: if the error can be confirmed, it will be a MUPDF problem ... but anyway, we need an example.

**Comment by JorjMcKie** [comment]
Reminder: this issue cannot be processed if you don't provide an example TIFF.

**Comment by leo20160311** [comment]
i found your email address：[jorj.x.mckie@outlook.de]，and  I am now sending it to you via email. Thank you 
Look forward to your reply

**Comment by JorjMcKie** [comment]
Thanks for submitting the image. It is a **_very_** small image. I was able to convert it to PDF. The result is not black but looks like so:
![grafik](https://user-images.githubusercontent.com/8290722/230896706-52f313c9-e9d4-4ff6-8fa6-43139d58fcec.png)
Don't know if this is correct or not ...

**Comment by leo20160311** [comment]
I'm very sorry, I provided you with an incorrect example
Yes, you are right. After zooming in on this TIFF file, I found that it matches the converted PDF file. I thought it was the same as what I tested on a virtual computer, but now it's not a pure black block.

I finally found the appropriate TIFF file and sent you a new email containing the TIFF file and my test results. Please help, thank you very much

**Comment by leo20160311** [comment]
Additionally, in my previous testing, I used pymupdf to convert the TIFF file of CCITT T.6 into PDF and save it to disk. When using Pillow to append the TIFF file of packbits  to the aforementioned PDF, Pillow's pdfreader reported an error and was unable to read the PDF file generated by pymupdf~

**Comment by leo20160311** [comment]
> Thanks for the new files. I tested them with the new PyMuPDF version coming out soon. Good news: only one of the example files has problems (convered to black): `gray.packbits.u2.tif`, all the others work correctly.

Thank you very much for your patient response. I am looking forward to the final result

**Comment by leo20160311** [comment]
@JorjMcKie @radarher 
Wow, is Radarrhere and JorjMcKie on the same team? First of all, thank you to the two enthusiastic technical experts

https://github.com/python-pillow/Pillow/issues/7084
This issue was also submitted by me. I tried some wrapping solutions, but I couldn't solve any problems, haha
My focus is: I have a bunch of images and documents in various formats, and I need to merge them into one PDF I haven't been able to achieve it yet, I need help, thank you very much

**Comment by leo20160311** [comment]
I just checked this file<gray.packbits.u2.tif> and realized that the content itself is black. Could it be that I misled you? However, it may be necessary to note that this file has two pages, both of which are black










At 2023-04-12 20:56:45, "Jorj X. McKie" ***@***.***> wrote:

Thanks for the new files. I tested them with the new PyMuPDF version coming out soon.
Good news: only one of the example files has problems (convered to black): gray.packbits.u2.tif, all the others work correctly.

**Comment by JorjMcKie** [comment]
According to our e-mail conversation, this issue can be closed with the new version, right?

**Comment by leo20160311** [comment]
I would like to test it with the new version so that it can be confirmed, thanks

**Comment by JorjMcKie** [comment]
> I would like to test it with the new version so that it can be confirmed, thanks

Ok, sure. Then please install the new version, test it and let us know your findings.
Thank you!

**Comment by leo20160311** [comment]
Thank you very much for your reply. You have completely resolved my issue through version 1.22.0. Thank you very much
